src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,289 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitIntercept,
+    const size_t max_iter,
+    const double tol,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+  Train(data, labels, max_iter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const double regularization,
+    const bool fitIntercept,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t max_iter,
+    const double tol)
+{
+  trainLabels = arma::Row<int> (data.n_cols);
```

Comment:
are these labels or coefficients?

Reply:
`Labels` is more suitable because using as label for every training sample.

Reply:
no but how is it a label? -1 is not a label for a given data point ?

Reply:
and then what is 0 and 1 labels that we receive from the user?

Reply:
Ok, I will change it your point is more valid.